app/src/main/kotlin/at/bitfire/davdroid/db/Collection.kt

```diff
@@ -202,6 +206,10 @@ data class Collection(
                 }
             }
 
+            if (timezoneId == null && timezone != null) {
```

Comment:
That just gives precedence to `calendar-timezone-id`, in case it's specified with `calendar-timezone`.

With that, if timezoneId is already set (by `calendar-timezone-id`), it will take that value; and if it's not set (and a timezone was specified), it loads the id from the full timezone